generatebundlefile/promote.go

```diff
@@ -120,29 +119,38 @@ func (c *SDKClients) PromoteHelmChart(repository, authFile string, crossAccount
 	if crossAccount {
 		destination = c.ecrPublicClientRelease
 	}
-	fmt.Printf("helmRequires.Spec.Images=%v\n", helmRequires.Spec.Images)
 	// Loop through each image, and the helm chart itself and check for existance in ECR Public, skip if we find the SHA already exists in destination.
 	// If we don't find the SHA in public, we lookup the tag from Private, and copy from private to Public with the same tag.
 	for _, images := range helmRequires.Spec.Images {
-		check, err := destination.shaExistsInRepository(images.Repository, images.Digest)
+		checkSha, err := destination.shaExistsInRepository(images.Repository, images.Digest)
+		checkTag, err := destination.tagExistsInRepository(images.Repository, version)
 		if err != nil {
 			return fmt.Errorf("Unable to complete sha lookup this is due to an ECRPublic DescribeImages failure %s", err)
 		}
-		if check {
-			BundleLog.Info("Image Digest already exists in destination location......skipping", images.Repository, images.Digest)
+		if checkSha && checkTag {
```

Comment:
What if only one or the other exists? Is it okay to execute the `else` branch in that case?

Reply:
Jonah caught me on slack to say:
> The copy is safe, and should run if only one check passes otherwise there will be helm container pull errors.